kge/job/train.py

```diff
@@ -252,18 +254,21 @@ def load(self, filename: str) -> str:
         Returns job id of the job that created the checkpoint."""
         self.config.log("Loading checkpoint from {}...".format(filename))
         checkpoint = torch.load(filename, map_location="cpu")
+        if self.config.get("job.type") == "train" and checkpoint["type"] == "package":
+            raise ValueError("Can not continue training on a packaged model")
         if "model" in checkpoint:
             # new format
             self.model.load(checkpoint["model"])
         else:
             # old format (deprecated, will eventually be removed)
             self.model.load_state_dict(checkpoint["model_state_dict"])
-        self.optimizer.load_state_dict(checkpoint["optimizer_state_dict"])
-        if "lr_scheduler_state_dict" in checkpoint:
-            # new format
-            self.kge_lr_scheduler.load_state_dict(checkpoint["lr_scheduler_state_dict"])
-        self.epoch = checkpoint["epoch"]
-        self.valid_trace = checkpoint["valid_trace"]
+        if checkpoint["type"] != "package":
```

Comment:
This should always be the case here.

Reply:
the trainjob.load method is also used if we call kge eval/test.
therefore we need this line to be able to load packaged models for evaluation and don't get complications with the optimizer, which we don't need

Reply:
We shouldn't use TJ.load for eval/test jobs: there is no need to load things like optimizer states. And making training jobs aware of other jobs is not a good design.